src/forms/components/encounter/ohri-encounter-form.tsx

```diff
@@ -113,12 +123,20 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
       setEncounterLocation(encounter.location);
     } else {
       allFormFields.forEach(field => {
-        if (field.questionOptions.rendering == 'checkbox') {
-          tempInitVals[field.id] = [];
-        } else if (field.questionOptions.rendering == 'toggle') {
-          tempInitVals[field.id] = false;
+        let value = null;
+        if (!isEmpty(field.questionOptions.defaultValue)) {
+          value = inferInitialValueFromDefaultFieldValue(field, encounterContext);
+        }
+        if (!isEmpty(value)) {
+          tempInitVals[field.id] = value;
         } else {
-          tempInitVals[field.id] = '';
+          if (field.questionOptions.rendering == 'checkbox') {
```

Comment:
Can you create a `key:value` structure that can hold the initialValues? something like

``` 
{
  'checkbox': [],
  'toggle': false,
  'default': ''
}
```

This will help reduce the number of `if / else` that you have

Reply:
Resolved